go-controller/pkg/ovn/address_set.go

```diff
@@ -0,0 +1,201 @@
+package ovn
+
+import (
+	"fmt"
+	"net"
+	"strings"
+	"sync"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	"k8s.io/klog"
+)
+
+type addressSet struct {
+	sync.Mutex
+	name     string
+	hashName string
+	uuid     string
+	ips      map[string]net.IP
+}
+
+// hash the provided input to make it a valid addressSet name.
+func hashedAddressSet(s string) string {
+	return hashForOVN(s)
+}
+
+type addressSetIterFn func(name, namespace, suffix string)
+
+// forEachAddressSetUnhashedName will pass the unhashedName, namespaceName and
+// the first suffix in the name to the 'iteratorFn' for every address_set in
+// OVN. (Each unhashed name for an addressSet can be of the form
+// namespaceName.suffix1.suffix2. .suffixN)
+func forEachAddressSetUnhashedName(iteratorFn addressSetIterFn) error {
+	output, stderr, err := util.RunOVNNbctl("--data=bare", "--no-heading",
+		"--columns=external_ids", "find", "address_set")
+	if err != nil {
+		return fmt.Errorf("error reading address sets: "+
+			"stdout: %q, stderr: %q err: %v", output, stderr, err)
+	}
+	for _, addrSet := range strings.Fields(output) {
+		if !strings.HasPrefix(addrSet, "name=") {
+			continue
+		}
+		addrSetName := addrSet[5:]
+		names := strings.Split(addrSetName, ".")
+		addrSetNamespace := names[0]
+		nameSuffix := ""
+		if len(names) >= 2 {
+			nameSuffix = names[1]
+		}
+		iteratorFn(addrSetName, addrSetNamespace, nameSuffix)
+	}
+	return nil
+}
+
+func NewAddressSet(name string, ips []net.IP) (*addressSet, error) {
```

Comment:
General comment, it would be beneficial to have unit test cases for the methods defined in this file

Reply:
@vjayaramrh now with address set object unit tests.